sforzando/q-lako
# List the search result with Amazon API

## Describe the PR

To close #11 .
List the search result with Amazon API.

## Screenshots

<img width="1904" alt="Screen Shot 2020-11-27 at 14 32 01" src="https://user-images.githubusercontent.com/67737076/100414578-5ea8b900-30bd-11eb-8e69-edfd28badf29.png">

## Detail of the change

1. main.py
    - Receive string from `POST /search`
    - Run `amazon_api_client.search_products(keywords=***)`
    - Format response data from Amazon PA-API to show as list
        - e.g. `products_list = [{"asin": aaaa, "title": bbbb, "image_url": cccc} , … , {"asin": xxxx, "title": yyyy, "image_url": zzzz}]
    - Run `render_template("search.html",  **context_dict)` with formatted data
1. search.html
    - Show data with Jinja template
        - Each asset is displayed with `<div>`  as shown below
    - Push `Select` button then request to `GET /registration` with asin code
1. registration.html, main.py
    - The string to be GETed is displayed as is

```html

{% for product in product_list %}
<div>
  <image src="{{ product.images.large }}" alt="image of {{ product.title }}" />
  <h2>{{ product.title }}</h2>
  <form action="/registration" method="post">
    <div>
      <input type="hidden" name="asin" value="{{ product.asin }}">
      <button>Select</button>
    </div>
  </form>
</div>
{%- endfor %}

```

## Anticipated impacts

Affects the `Amazon PA-API` call limits.

## To reproduce

Steps to check the behavior:

1. Run `python main.py`
1. Visit to `http://0.0.0.0:8888/`
1. Enter one of the following keywords: keyword, ISBN code, or ASIN code
1. Make sure you see the list of items as in the screenshot
1. Press the "Select" button
1. Check see the ASIN code of the selected item

## Additional context

Next issue is #13
